### DISCUSSION TITLE
PyVista ImGui Repo

### DISCUSSION THREAD

**Discussion by adam-grant-hendry** [pyvista/pyvistaqt#208] [Polls]
@akaszynski @tkoyama010 @larsoner @adeak I like [pyimgui](https://pyimgui.readthedocs.io/en/latest/). As an [immediate mode renderer](https://en.wikipedia.org/wiki/Immediate_mode_(computer_graphics)), it has performance benefits over [retained mode renderers](https://en.wikipedia.org/wiki/Retained_mode) and can make some pretty great interfaces.

`Qt`/`PyQt`/`PySide` are a bit trickier to work with and have had licensing woes and variations across bindings that have caused headaches. [Dear imgui](https://github.com/ocornut/imgui) is open-source MIT.

If you agree, I'd like to take a crack at combining it with `pyvista` and `vtk` [like was done here](https://discourse.vtk.org/t/vtk-dearimgui-integration-through-vtkgenericopenglrenderwindow/4424). What do you think?

**Comment by adeak** [comment]
I don't want to vote due to my usual disclaimer of not being a pyvistaqt (or any other GUI) user. Just a few musing remarks:
1. It's harmless to experiment, so if you have time to invest/waste you can see how it works for you. With an existing (even half-complete) prototype it might be easier to weigh pros and cons. Just make it clear that the repo is an experimental work in progress and is not meant to be used (yet).
2. Are the Qt stuff retained mode renderers then? If yes, how would the paradigm shift impact performance of apps, especially with or without dedicated GPUs?
3. `pyimgui` might have some maintenance issues: [the repo](https://github.com/pyimgui/pyimgui) has a note saying that `master` is obsolete, `dev/version-2.0` prepares a new major version for newer `DearImGui`, but both branches were last committed to in April. I don't know how the Qt dependencies are maintained, but this makes this seem a bit risky.

**Reply by tkoyama010** [reply]
> It's harmless to experiment, so if you have time to invest/waste you can see how it works for you. With an existing (even half-complete) prototype it might be easier to weigh pros and cons. Just make it clear that the repo is an experimental work in progress and is not meant to be used (yet).

FYI. It is possible to create a repository on your account and later move it to the PyVista organization. When I create an experimental repository, I first create it under my own account and then move it to the PyVista organization once it is stable.

**Reply by akaszynski** [reply]
Sure! I agree with (1-3) from @adeak, but I can't promise support on my end as I'm swamped with real work as well as pyvista work.
